src/components/PopupWithDelete.js

```diff
@@ -0,0 +1,27 @@
+import Popup from "./Popup";
+
+export default class PopupWithDelete extends Popup {
+  constructor(popupSelector) {
+    super(popupSelector);
+
+    this._deleteCardForm =
+      this._popupElement.querySelector(".modal__container");
+
+    this._deleteCardButton = this._popupElement.querySelector(".modal__button");
+  }
+
+  setEventListeners() {
+    super.setEventListeners();
+
+    this._deleteCardForm.addEventListener("submit", (evt) => {
+      evt.preventDefault();
+      this._deleteCardButton.textContent = "Saving...";
```

Comment:
Needs correction: you should start the loading only before the api request in index.js as you don't know how the form submission is processed

Also you should create a method for rendering loading, so the logic won't be duplicated

One of the options to implement that would be adding a parameter `loadingButtonText` to the constructor of `PopupWithForm` and using it to change button text. 

```js
constructor(popupSelector, { handleSubmit, loadingButtonText }) { 
  ...
  this._submitButton = this._formElement.querySelector(...)
  this._buttonText = this._submitButton.textContent;
  this._loadingButtonText = loadingButtonText;
  …
}

showLoading() {
  this._submitButton.textContent = this._loadingButtonText
}

hideLoading() {
  this._submitButton.textContent = this._buttonText
}
``` 

The api call should look like this:

```js
popup.showLoading()
api.
  .then(...)
  .catch(...)
  .finally(() => {
    popup.hideLoading()
```